src/core/sagas/products.ts

```diff
@@ -9,26 +11,71 @@ import * as utils from '../utils';
 import { Tasks as productDetailsTasks } from './product-details';
 
 export namespace Tasks {
-  export function* fetchProducts(flux: FluxCapacitor, action: Actions.FetchProducts) {
+  export function* fetchProductsAndNavigations(flux: FluxCapacitor, action: Actions.FetchProducts) {
```

Comment:
should just keep the name as `fetchProducts()` as it will only become more complex as time goes on, and it's primary function is to fetch the products, whereas it will only request navigations conditionally